autogen/agentchat/conversable_agent.py

```diff
@@ -411,17 +411,54 @@ def _summary_from_nested_chats(
         res = initiate_chats(chat_to_run)
         return True, res[-1].summary
 
+    @staticmethod
+    async def _a_summary_from_nested_chats(
+        chat_queue: List[Dict[str, Any]], recipient: Agent, messages: Union[str, Callable], sender: Agent, config: Any
+    ) -> Tuple[bool, str]:
+        """A simple chat reply function.
+        This function initiate one or a sequence of chats between the "recipient" and the agents in the
+        chat_queue.
+
+        It extracts and returns a summary from the nested chat based on the "summary_method" in each chat in chat_queue.
+
+        Returns:
+            Tuple[bool, str]: A tuple where the first element indicates the completion of the chat, and the second element contains the summary of the last chat if any chats were initiated.
+        """
+        last_msg = messages[-1].get("content")
+        chat_to_run = []
+        for i, c in enumerate(chat_queue):
+            current_c = c.copy()
+            if current_c.get("sender") is None:
+                current_c["sender"] = recipient
+            message = current_c.get("message")
+            # If message is not provided in chat_queue, we by default use the last message from the original chat history as the first message in this nested chat (for the first chat in the chat queue).
+            # NOTE: This setting is prone to change.
```

Comment:
What setting is "prone to change"?  It's unclear from this comment.

Reply:
This is essentially a copy-paste of `summary_from_nested_chats` following how other async methods are implemented. I'm happy to refactor a bit so most of the logic is shared between the two functions.

Reply:
Didn't realize this a copy and paste, makes more sense in that context, thank you.